### PR TITLE
fix(MenuItem): Prevent to dismiss menu opened on hover

### PR DESCRIPTION
**Pull request by assuncaocharles**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [ ] Include a change request file using `$ yarn change`

#### Description of changes

Currently if the menu item that contains a menu get clicked it closes the menu, this PR prevents this to happen

#### Focus areas to test

(optional)

### CODE REVIEW

File: packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx [hunk 1]
```diff
@@ -320,7 +320,7 @@ export const MenuItem = compose<'a', MenuItemProps, MenuItemStylesProps, {}, {}>
           trySetMenuOpen(false, e, () => focusAsync(itemRef.current));
         } else {
           // the menuItem element was clicked => toggle the open/close and stop propagation
-          trySetMenuOpen(active ? !menuOpen : true, e);
+          trySetMenuOpen(active && on !== 'hover' ? !menuOpen : true, e);
```
Comment (pompomon) [inline packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:323]: Could you add a test for that, please? :)

Comment (assuncaocharles) [reply packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx:323]: Done

**Review by pompomon — APPROVED** [review]

File: packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenuHover.shorthand.tsx [hunk 2]
```diff
@@ -8,7 +8,6 @@ const items: MenuProps['items'] = [
     on: 'hover',
     menu: {
       items: [
-        { key: '1', content: 'item1' },
```
Comment (layershifter) [inline packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenuHover.shorthand.tsx:11]: Why this was removed?

Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenuHover.shorthand.tsx:11]: No reason, there was no need to have it there only

Comment (assuncaocharles) [reply packages/fluentui/docs/src/examples/components/Menu/Usage/MenuExampleWithSubmenuHover.shorthand.tsx:11]: Reverted

File: packages/fluentui/react-northstar/test/specs/components/Menu/Menu-test.tsx [hunk 3]
```diff
@@ -54,6 +54,37 @@ describe('Menu', () => {
...
+          .at(0)
+          .find('a')
+          .first()
+          .getDOMNode()
+          .getAttribute('aria-expanded'),
+      ).toBe('false');
+
+      menuItems
+        .at(1)
+        .simulate('mouseover')
```
Comment (pompomon) [inline packages/fluentui/react-northstar/test/specs/components/Menu/Menu-test.tsx:74]: Please should we have a separate check after `mouseover` and after `click`? To see that menu actually opened on hover and kept opened after the click?

**Review by pompomon — APPROVED** [review]

**Pull request merged**